### PR TITLE
feat(WRN-1350): support gradle runinstall

### PR DESCRIPTION
**Pull request by PhilipAbed**

### CODE REVIEW

**Review by nabeelsaabna — CHANGES_REQUESTED** [review]
add/improve Docs
add JsDocs for "types"

**Review by nabeelsaabna — CHANGES_REQUESTED** [review]

File: src/tools/gradle.js [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+    return undefined;
+}
+
+/**
+ * @typedef {Object} ToolConstraint
+ * @property {string} source - The source of the tool, e.g., 'gradle'.
+ * @property {string} constraint - The constraint associated with the tool.
+ * @property {string} toolName - The name of the tool, e.g., 'gradle'.
+ */
```
[the remark this answers is not in the document]
Comment (PhilipAbed) [reply src/tools/gradle.js:72]: object type of ToolConstraint. is there another way to do it ? :D

File: README.md [hunk 2]
```diff
@@ -1,12 +1,26 @@
 # Runinstall
 
+Runinstall is a tool that detects the project version and installs it dynamically.
+
+Runinstall has the propriety code to extract the constraints from the supported tools, and it utilizes renovate and install-tool in order to generate commands and install them.
```
Comment (PhilipAbed) [inline README.md:5]:
```suggestion
Runinstall extracts version constraints of package managers, and utilizes Renovate and install-tool to generate commands to install the tool version that the project requires.
```

[the remark this answers is not in the document]
Comment (PhilipAbed) [reply README.md]: try now

**Review by nabeelsaabna — APPROVED** [review]
Docs to be refined in a new PR since it's not a Gradle related change

**Pull request merged**